extension/llm/export/builder.py

```diff
@@ -166,6 +171,37 @@ def capture_pre_autograd_graph(self) -> "LLMEdgeManager":
             )
         return self
 
+    def calibrate(
+        self,
+        prepared_module,
+        calibration_tasks,
+        calibration_limit,
+        calibration_seq_length,
+        tokenizer_path,
+    ):
+        logging.info("run calibration...")
+        try:
+            from executorch.examples.models.llama2.evaluate import EagerEvalWrapper
+        except ImportError:
+            raise ImportError(
+                "Please install the llm eval dependency via examples/models/llama2/install_requirements.sh"
+            )
+
+        tokenizer = get_tokenizer(tokenizer_path)
+        eval_wrapper = EagerEvalWrapper(
```

Comment:
Thanks for this change.
It seems to have a problem when I capture seq_len = 1 but [calibrate with whole sequence in batch process](https://github.com/pytorch/executorch/blob/d59419c4d56f7f14b3e6ac65848ce23c1b3ee108/examples/models/llama2/evaluate/eager_eval.py#L85).
From my understanding, after we capture, some variable will be fixed such as ["batch, seqlen = x.shape"](https://github.com/pytorch/executorch/blob/d59419c4d56f7f14b3e6ac65848ce23c1b3ee108/examples/models/llama2/llama_transformer.py#L315).

Reply:
Hi thank you for checking out the pr! I pulled some changes from another local patch and haven't tested this pr properly. What you mentioned is true...I vaguelly remember I modified the code there to use kv cache instead, however it was too slow to calibrate (as you observed too)